pkg/provision/engine.go

```diff
@@ -182,6 +182,19 @@ func (e *Engine) provision(ctx context.Context, r *Reservation) error {
 		return errors.Wrapf(err, "failed to cache reservation %s locally", r.ID)
 	}
 
+	// If an update occurs on the network we don't increment the counter
+	if r.Type == "network" {
+		nr := pkg.NetResource{}
+		if err := json.Unmarshal(r.Data, &nr); err != nil {
+			return fmt.Errorf("failed to unmarshal network from reservation: %w", err)
+		}
+
+		exists, _ := e.cache.NetworkExists(nr.Name, r.User)
```

Comment:
Not really because this function contains a known issue now. It's better to check if the name and user are equal no?

Reply:
Using this now. This method also addresses the issue in https://github.com/threefoldtech/zos/issues/898. The reasons it's defined in engine.go is because we get an import cycle if we define it in primitives.

https://github.com/threefoldtech/zos/blob/e729d31d50a0eda533cc359d3f58c97bbe40ee85/pkg/provision/engine.go#L398-L408

Reply:
using this now https://github.com/threefoldtech/zos/pull/897#discussion_r474521613